File: app/src/main/java/com/dominikgaller/alpinebooking/booking/core/outport/AvailabilityUnavailableException.java
package com.dominikgaller.alpinebooking.booking.core.outport;

/**
 * Thrown by {@link AvailabilityChecker} when the availability source cannot be reached
 * or returns an invalid response.
 *
 * <p>Maps to HTTP 502 Bad Gateway at the REST boundary.
 *
 * <p>SDD: See {@code documentation/ports/availability-checker.outport.spec.md}, section 2.1.
 */
public class AvailabilityUnavailableException extends RuntimeException {

    public AvailabilityUnavailableException(final String message) {
        super(message);
    }

    public AvailabilityUnavailableException(final String message, final Throwable cause) {
        super(message, cause);
    }
}

File: src/main/kotlin/com/example/service/booking/core/domain/tourbooking/exception/AvailabilityUnavailableException.kt
package com.example.service.booking.core.domain.tourbooking.exception

/**
 * Thrown when the availability system cannot be reached or answers unusably.
 *
 * This is not a domain rule being violated — it is a dependency failing — so it maps to
 * HTTP 502 rather than to 400 or 409.
 */
class AvailabilityUnavailableException(
    message: String,
    cause: Throwable? = null,
) : RuntimeException(message, cause)
